t/lib/Test/Act/Environment.pm

```diff
@@ -0,0 +1,221 @@
+BEGIN {
+    $tempdir = File::Temp->newdir( CLEANUP => 0 );
+    # The following will need adaption to the post-november master
+    # Part 1: Link the distribution files from the repository so that
+    # changes in the working tree become immediately effective
+    for my $dir (qw(templates po wwwdocs)) {
+        symlink "$RealBin/../$dir","$tempdir/$dir"
+            or die "Failed to create a symlink for '$RealBin/../$dir': '$!'";
+    }
+    # Part 2: Copy the files from the test environment
+    for my $dir (qw(actdocs conf conferences)) {
+        dircopy "$RealBin/acthome/$dir","$tempdir/$dir";
+    }
+    # Part 3: These must just exist
+    for my $dir (qw(photos ttc)) {
+        mkdir "$tempdir/$dir" or die "Could not create '$tempdir/$dir': '$!'";
+    }
+    $ENV{ACTHOME} = "$tempdir";
+}
+
+use Act::Config;
+
+my $smtp_server = Test::Act::SMTP::Server->instance;
+$Config->set(email_hostname => 'localhost');
+$Config->set(email_port     => $smtp_server->port);
+
+
+has mech => (
+    is => 'rwp', isa => InstanceOf['Test::WWW::Mechanize'],
+    builder => 'build_mech',
+    documentation =>
+        'A test client for the application tests',
+);
+
+sub build_mech ($self) {
+    require Act::Dispatcher;
+    my $mech = Test::WWW::Mechanize::PSGI->new(app => Act::Dispatcher->to_app);
+    $self->_set_mech($mech);
+    return $mech;
+}
+
+has base => (
+    is => 'ro', isa => Str,
+    builder => '_build_base',
+    documentation =>
+        'The base URL for tests',
+);
+
+sub _build_base ($self) {
+    my $host = `hostname`;  chomp $host;
+    my $port = $ENV{ACT_TEST_PORT} || 5050;
+    return "http://$host:$port";
+}
+
+has home => (
+    is => 'ro', isa => Str,
+    default => sub { "$tempdir" },
+    documentation =>
+        'Where $ENV{ACTHOME} will point to',
+);
+
+sub _build_acthome ($self) {
+    my $tempdir = File::Temp->newdir( CLEANUP => 0 );
+    # The following will need adaption to the post-november master
+    # Part 1: Link the distribution files from the repository so that
+    # changes become immediately visible
+    for my $dir (qw(templates po wwwdocs)) {
+        symlink "$RealBin/../$dir","$tempdir/$dir"
+            or die "Failed to create a symlink for '$RealBin/../$dir': '$!'";
+    }
+    # Part 2: Copy the files from the test environment
+    for my $dir (qw(actdocs conf)) {
+        dircopy "$RealBin/acthome/$dir","$tempdir/$dir";
+    }
+    # Part 3: These must just exist
+    for my $dir (qw(photos ttc)) {
+        mkdir "$tempdir/$dir" or die "Could not create '$tempdir/$dir': '$!'";
+    }
+    $ENV{ACTHOME} = $tempdir;
+    return "$tempdir";
+}
+
+has smtp_server => (
+    is => 'ro', isa => InstanceOf['Test::Act::SMTP::Server'],
+    default => sub { $smtp_server },
+    documentation =>
+        'The (singleton) SMTP service for this test run',
+);
+
+
+# ----------------------------------------------------------------------
+
+sub new_mech ($self) {
```

Comment:
What is the difference with `build_mech`? This seems to do the exact same thing. 

I am reading the docs now. Perhaps use a clearer on the `mech` attribute and then just let the builder rebuild the mech object.

Reply:
Yep, I'll fix that.